File: day04/cypher.py
def advance_three_letters(string):
  alpha = 'abcdefghijklmnopqrstuvwxyz'
  advanced = ''

  for s in string:
    if s in alpha:
      for a in range(len(alpha)):
        if alpha[a] == s and a <= 22:
          advanced += alpha[a+3]
        elif alpha[a] == s:
          advanced += alpha[a - 23]
    else:
      advanced += s

  print(advanced)

def get_string():
  string = input('Please enter a string of letters & words: ')

  try:
    advance_three_letters(string)
  except Exception as error:
    print('Error:', error)
    get_string()

if __name__ == '__main__':
  get_string()

File: day04/cypher.js
function advanceThreeLetters(string) {
  var alpha = 'abcdefghijklmnopqrstuvwxyz';
  var caps = alpha.toUpperCase();
  advanced = ''
  
  for (var s in string) {
    if (alpha.includes(string[s]) || caps.includes(string[s])) {
      for (var a in alpha) {
        if (alpha[a] == string[s] && a <= 22) {
          advanced += alpha[Number(a) + 3];
        } else if (alpha[a] == string[s]) {
          advanced += alpha[Number(a) - 23];
        } else if (caps[a] == string[s] && a <= 22) {
          advanced += caps[Number(a) + 3];
        } else if (caps[a] == string[s]) {
          advanced += caps[Number(a) - 23];
        }
      }
    } else {
      advanced += string[s]
    }
  }
  
  return advanced;
}

function getWordsToReverse() {
  var words = prompt('Please enter a string of words to reverse:');
  
  try {
    advanceThreeLetters(words);
  } catch(error) {
    console.log('Error:', error);
  }
}
